hydroflow_plus_test/src/cluster/paxos_bench.rs

```diff
@@ -378,31 +379,30 @@ fn bench_client<'a>(
                     median_latency_window_size
                 ))),
                 0usize,
-                false
             )),
-            q!(move |(latencies, write_index, has_any_value), latency| {
+            q!(move |(latencies, write_index), latency| {
                 let mut latencies_mut = latencies.borrow_mut();
-                if let Some(latency) = latency {
-                    // Insert into latencies
-                    if let Some(prev_latency) = latencies_mut.get_mut(*write_index) {
-                        *prev_latency = latency;
-                    } else {
-                        latencies_mut.push(latency);
-                    }
-                    *has_any_value = true;
-                    // Increment write index and wrap around
-                    *write_index += 1;
-                    if *write_index == median_latency_window_size {
-                        *write_index = 0;
-                    }
+                if *write_index < latencies_mut.len() {
+                    latencies_mut[*write_index] = latency;
                 } else {
-                    // reset latencies
-                    latencies_mut.clear();
-                    *write_index = 0;
-                    *has_any_value = false;
+                    latencies_mut.push(latency);
                 }
+                // Increment write index and wrap around
+                *write_index = (*write_index + 1) % median_latency_window_size;
             }),
         );
+
+    let median_latency = c_latencies.map(q!(|(latencies, _)| {
+        let mut latencies_mut = latencies.borrow_mut();
+        if latencies_mut.len() > 0 {
+            let middle_idx = latencies_mut.len() / 2;
+            let (_, median, _) = latencies_mut.select_nth_unstable(middle_idx);
```

Comment:
I believe, in this implementation, `select_nth_unstable` will reorder the elements of `c_latencies` every tick, instead of only when we want to calculate the median latency. I think it's *technically* safe...? But kind of odd behavior, since now the window doesn't necessarily include the last k latencies

Reply:
Oh good point, didn't think about this messing up the rolling window. I'll probably just revert this for now, the main reason I started working on this is there are occasional crashes at startup somewhere in the latency measurement calculations.